src/TestSuite/IntegrationTestCase.php

```diff
@@ -660,6 +661,35 @@ protected function _addTokens($url, $data)
         return $data;
     }
 
+    /**
+     * Recursively casts all data to string as that is how data would be POSTed in
+     * the real world
+     *
+     * @param array $data POST data
+     * @return array
+     */
+    protected function _castToString($data)
+    {
+        foreach ($data as $key => $value) {
+            if (is_scalar($value)) {
+                $data[$key] = (string)$value;
+
+                continue;
+            }
+
+            if (is_array($value)) {
+                $looksLikeFile = isset($value['error']) && isset($value['tmp_name']) && isset($value['size']);
+                if ($looksLikeFile) {
+                    continue;
+                }
+
+                $data[$key] = $this->_castToString($value);
+            }
+        }
```

Comment:
This might be a good place to use [array_walk_recursive](http://php.net/array_walk_recursive)

Reply:
I don't think that would work well given the check for a file array, as `array_walk_recursive` is unaware of its sibling keys afaik.

Reply:
I think the code is readable as of now.